libs/db.lua

```diff
@@ -289,6 +289,10 @@ return function (path)
 
       for entry in assert(fs.scandir(path)) do
         local fullPath = pathJoin(path, entry.name)
+        if entry.type == nil then
+          local stat = fs.stat(fullPath)
+          entry.type = stat.type
+	    end
```

Comment:
In lua, `and` and `or` implement short-circuit and return the last value, so this could be the one-liner:

```lua
entry.type = entry.type or fs.stat(fullPath).type
```